src/java.base/share/classes/sun/security/ssl/SSLEngineImpl.java

```diff
@@ -760,11 +760,12 @@ private Plaintext decode(
     @Override
     public Runnable getDelegatedTask() {
         engineLock.lock();
+        HandshakeContext context = conContext.handshakeContext;
         try {
-            if (conContext.handshakeContext != null && // PRE or POST handshake
+            if (context != null && // PRE or POST handshake
                     !conContext.handshakeContext.taskDelegated &&
                     !conContext.handshakeContext.delegatedActions.isEmpty()) {
```

Comment:
Shouldn't replace `conContext.handshakeContext` with `context` in those two lines as well?

Reply:
Yes, I definitely should have done that. Thanks.